src/api/app/views/webui2/webui/staging/workflows/_problems.html.haml

```diff
@@ -1 +1,31 @@
-Placeholder for the content
+:ruby
+  max_problems = 5
+  project_class = project.name.tr(':', '_')
+  item_project_class = 'd-none hidden-item-' + project_class
+
+- if project.problems.empty?
+  .text-center
+    %i.fas.fa-check-circle.fa-2x.text-success
+- else
+  %ul.list-group.list-group-flush
+    - project.problems.each.with_index(1) do |(name, states), index|
+      %li.list-group-item.table-list-group-item{ class: "#{item_project_class if index > max_problems}" }
+        = link_to("#{name}:", package_show_path(package: name, project: project))
+        - states.each do |state, values|
+          = state
+          = surround '(', ')' do
+            - values.each do |value|
+              - coma = ', ' unless values.last == value
+              = succeed coma do
+                - if state == 'unresolvable'
+                  = value[:arch]
+                - else
+                  = link_to(value[:arch], package_live_build_log_path(package: name,
+                                                                      repository: value[:repository],
+                                                                      project: project,
+                                                                      arch: value[:arch]))
+    - length = project.problems.length
+    - number = length - max_problems
```

Comment:
I would argue that 'number' is only a little bit better than 'p' :)

And it only needs assigning in the if

Reply:
no, it is used twice

Reply:
both lines are in the if

Reply:
:see_no_evil: